test/unit/batch_alloc.c

```diff
@@ -0,0 +1,96 @@
+#include "test/jemalloc_test.h"
+
+#define BATCH ((1U << 16) + 101)
+
+void
+verify_batch(void **ptrs, size_t batch, size_t usize, bool zero) {
+	tsdn_t *tsdn = tsdn_fetch();
+	szind_t ind = sz_size2index(usize);
+	const bin_info_t *bin_info = &bin_infos[ind];
+	const unsigned nregs = bin_info->nregs;
+	void *ref = calloc(1, usize);
+	assert_ptr_not_null(ref, "");
+	for (size_t i = 0, j = 0; i < batch; ++i, ++j) {
+		if (j == nregs) {
+			j = 0;
+		}
+		void *p = ptrs[i];
+		expect_zu_eq(isalloc(tsdn, p), usize, "");
+		if (zero) {
+			expect_d_eq(memcmp(p, ref, usize), 0, "");
+		}
+		free(p);
+		if (j == 0) {
+			expect_true(prof_sample_aligned(p), "");
```

Comment:
We should be clearer about our expectations for prof_sample alignment vs. merely page alignment. We should be able to turn up the alignment of profiled objects without breaking tests.